ecc_enclave/enclave/shim.h

```diff
@@ -33,20 +33,69 @@ int invoke(uint8_t* response,
 
 // put/get state
 //-------------------------------------------------
+
+// Normal, encrypted state
+
 // - store value located at val of size val_len under key key
+//   Note:
+//   - while the values are encrypted, the key will remain in clear text.
+//     So care has to be taken by the programmer that the key doesn't leak
+//     anything sensitive!
 void put_state(const char* key, uint8_t* val, uint32_t val_len, shim_ctx_ptr_t ctx);
+
 // - look for key and, if found, store it in val and return size in val_len.
 //   val must be of size at least max_val_len and the query will fail
 //   if the retrieved value would be larger.
 //   Absence of key is denoted by val_len == 0 when the function returns.
+//   Note:
+//   - this function doesn't check whether this was also stored privately.
+//     If it was stored with put_public_state, reading the key will fail
+//     due to an integrity failure.
 void get_state(
     const char* key, uint8_t* val, uint32_t max_val_len, uint32_t* val_len, shim_ctx_ptr_t ctx);
+
 // - look for composite keys, i.e., return the set of keys and values which match the
 //   provided composite (prefix) key comp_key
+//   Note:
+//   - this function doesn't check whether this was also stored privately.
```

Comment:
you mean.. "publicly"?

Reply:
nope, i meant privately.  I guess either would have worked (in my case, as a test for consistency; in your case as a case of inconsistency ...